### PR TITLE
Only PURGE the_permalink on clean_post_cache

### PR DESCRIPTION
**Pull request by joshbetz**
For normal post updates (publish->publish), we don't need to clear all related URLs because the post still exists. Only clear all the related URLs when the status transitions (i.e. draft->publish or publish->draft).

### CODE REVIEW

**Comment by nickdaugherty** [comment]
My thoughts:

This is going to leave stale data in places like feeds and archive pages - for example, if a post title has a typo and it's noticed after the post is live, editorial will go to fix it, but the update won't be reflected in feeds, archives, etc, for up to 30m and folks are going to wonder why.

**Comment by joshbetz** [comment]
I think the alternative is to lower the TTL on feeds to 5 minutes and disable purges, which will lead to more stale data.

**Comment by nickdaugherty** [comment]
IMO the right way to handle this is to aggregate purge requests across a given timeframe, so that we can't spam an arbitrary number of purges against the caches, which is what happens currently with no throttling.

File: cache-manager.php [hunk 1]
```diff
@@ -246,11 +248,65 @@ function queue_post_purge( $post_id ) {
...
+			return;
+		}
+
+		// Only send PURGE requests for public post types
+		if ( ! is_post_type_viewable( $post->post_type ) ) {
+			return;
+		}
+
+		if ( 'publish' !== get_post_status( $post_id ) ) {
+			return
```
Comment (nickdaugherty) [inline cache-manager.php]: Missing semi-colon

File: cache-manager.php [hunk 2]
```diff
@@ -246,11 +248,65 @@ function queue_post_purge( $post_id ) {
...
+			return;
+		}
+		
+		// Only send PURGE requests for public post types
+		if ( ! is_post_type_viewable( $post->post_type ) ) {
+			return;
+		}
+		
+		// Don't PURGE if we're not transitioning to or from publish
+		if ( $new === $old || ( $new != 'publish' && $old != 'publish' ) ) {
```
Comment (nickdaugherty) [inline cache-manager.php]: Do other post statuses need to be considered here? Mostly thinking about EditFlow.

File: cache-manager.php [hunk 3]
```diff
@@ -246,11 +248,65 @@ function queue_post_purge( $post_id ) {
...
+		// PURGE all feeds if one of the whitelisted fields is updated
+		foreach( $fields as $field ) {
+			if ( $post->{$field} != $old_post->{$field} ) {
+				$this->clear_feed_urls( $post );
+				return;
+			}
+		}
+	}
+
+	function clear_feed_urls( $post ) {
```
Comment (nickdaugherty) [inline cache-manager.php]: We may want to name this function differently, since it purges more than just feeds.

File: cache-manager.php (hunk 2 above)
Comment (joshbetz) [inline cache-manager.php]: I thought about that... I think publish is always publish. Edit Flow adds intermediate steps before publish, but (afaik) the publish state is still the same.

Comment (nickdaugherty) [inline cache-manager.php]: Cool - that's what I was thinking too. If we find there are edge cases, we can revisit.

**Comment by joshbetz** [comment]
Regardless of additional measures, this should only eliminate unnecessary cache purges.

**Pull request closed without merging**